src/algorithm.rs

```diff
@@ -135,18 +135,21 @@ impl<N: Network + 'static, K: KVStore, A: App<K> + 'static> Algorithm<N, K, A> {
 
             // 4. In case the view has been updated, update HotStuff's internal view and perform
             // the necessary protocol steps.
-            self.hotstuff.on_receive_view_info(view_info.clone(), &mut self.block_tree, &mut self.app);
+            if view_info.view != self.hotstuff.view_info().view {
```

Comment:
I wonder whether it'd be more decoupled if checking whether the `ViewInfo` has changed is moved into `on_receive_new_view` instead of being done in `algorithm`. This way, we won't have to expose this piece of `HotStuff` internal state to the outside through the `view_info` getter.

Reply:
True, though I can see advantage in it being explicit that this method is only called when the view changes. I have a different concern regarding this piece of code. I don't like the name `on_receive_view_info`, because it is similar to the names of methods called on received messages, but `view_info` is not a message. I'd rather rename it to `on_update_view_info(new_view_info: ViewInfo...)` and keep the current condition. This way it is more explicit what is happening, matching the clarity offered by the previous `execute_view` method. What do you think?

Also note that with `hotstuff.view_info()` we are calling a public hotstuff method, not accessing its state.

Reply:
I am okay with either name. And yes, I am aware that `view_info` is a getter method (I noted that it is a getter in my original comment), but I think this is not much better than accessing the field directly, which is to say that it to me is still undesirable coupling. Refer to [guideline A1.1](https://github.com/parallelchain-io/l1-team-engineering-guidelines?tab=readme-ov-file#a11-interact-through-public-interfaces-not-state-mutations).

Reply:
Sure, one alternative is to have hotstuff method called `is_view_outdated(pacemaker_view_info)`, that can be called instead of `view_info()`. I believe with this we could address both my and your concerns.